### PR TITLE
jsi改造与mmkv

### PR DESCRIPTION
**Pull request by FeiXie8**
Before submitting a new pull request, please make sure:

- [ ] Test cases have been added/updated/passed for the code you will submit.
- [ ] Documentation has added or updated.
- [ ] Commit message is following the [Convention Commit](https://conventionalcommits.org/) guideline with maximum 72 characters.
- [ ] Squash the repeat code commits, short patches are welcome.

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/Hippy?pullRequest=958) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/Hippy?pullRequest=958) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/Hippy?pullRequest=958) it.</sub>

File: android/sdk/src/main/jni/mmkvModule/cpp_adapter.h [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#pragma once
+#include <jni.h>
+#include <jsi/jsi.h>
+#include "core/mmkv/MMKV.h"
+
+namespace hippy{
+    namespace bridge{
+        void install(jsi::Runtime& jsiRuntime);
+        std::string jstringToStdString(JNIEnv *env, jstring jStr);
```
Comment (churchill-zhang) [inline android/sdk/src/main/jni/mmkvModule/cpp_adapter.h:9]: hippy原本的utils已经提供了工具函数

File: android/sdk/src/main/jni/include/bridge/js2javaJSI.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+#pragma once
+
+#include <jni.h>
+#include "jsi/jsi.h"
+#include <unordered_map>
+
+namespace hippy{
+namespace bridge{
+using namespace hippy;
```
Comment (churchill-zhang) [inline android/sdk/src/main/jni/include/bridge/js2javaJSI.h:31]: .h文件不推荐using namespace

File: android/sdk/src/main/jni/include/bridge/js2javaJSI.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	public:
+		static void install(
+			jsi::Runtime &runtime,
+			const std::shared_ptr<TurboModule> turboModule,
+			const std::string moduleName_
+		);
+	
+		
+	private:
+		JNIEnv jniEnv_;
```
Comment (churchill-zhang) [inline android/sdk/src/main/jni/include/bridge/js2javaJSI.h:48]: JNIEnv 不推荐存储

File: android/sdk/src/main/jni/include/jsi/jsi-inl.h [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+namespace hippy {
+namespace bridge {
+namespace detail {
+
+inline Value toValue(Runtime&, std::nullptr_t) {
```
Comment (churchill-zhang) [inline android/sdk/src/main/jni/include/jsi/jsi-inl.h:14]: Hippy 有对应的 NAPI 的封装，融合 rn 的 jsi 需要考虑直接使用 Hippy 的 NAPI

File: android/sdk/src/main/jni/src/bridge/entry.cc [hunk 5]
```diff
@@ -33,7 +33,16 @@
 #include <unordered_map>
 
 #include "bridge/java2js.h"
+
+#ifndef JSI
```
Comment (churchill-zhang) [inline android/sdk/src/main/jni/src/bridge/entry.cc:37]: JSI 本身 和 bridge 并不冲突，理论两者都可以共存，应对不同的业务场景

File: android/sdk/src/main/jni/src/bridge/entry.cc [hunk 6]
```diff
@@ -33,7 +33,16 @@
...
 #include "bridge/js2java.h"
+#else 
+#include "bridge/js2javaJSI.h"
+#ifdef MMKV
```
Comment (churchill-zhang) [inline android/sdk/src/main/jni/src/bridge/entry.cc:41]: MMKV 并没有看出来有绝对的必要性，但是增加了不少包大小

**Pull request closed without merging**